steelthread-0/walkthrough.md

```diff
@@ -15,123 +18,234 @@ The installation has already been performed on an OpenShift cluster as per the i
+- In the method select ```POST``` and click next
+- Again under `Select Type` set it to ```JSON Schema``` 
+- Add the following schema to the `Definition` field & click `Done`
+```
+{
+	"$schema": "http://json-schema.org/draft-04/schema#",
+	"type": "object",
+	"properties": {
+		"name": {
+			"type": "string"
+		},
+		"stock": {
+			"type": "string"
+		}
+	}
+}
+```
+
+- Next click the small plus between the two connections on the left hand side then click ```Add a step```
+- Click ```Data Mapper``` 
+- Map the `stock` property in the Source to the `stock` property in the Target by dragging from one side to the other. If done correct, a blue line will link the Source & Target.
+- Map `type` in the Source to `name` in the Target, again by dragging from one side to the other
+- Click `Done` at the top right
+
+## Add some Integration logging steps (optional)
+
+To add some more visibility we can add some logging steps to our integration.
+
+- On the left hand side inbetween the connections you will see small blue boxes with plus symbols. For each of these do the following:
+
+- click the blue plus symbol
+- click `Add a step`
+- choose `Log`
+- select ```Message Body```
+
+Once done we should now have 3 steps (5 if you added logging) in our integration. Click `Publish` in the top right.
+
+Name the integration ```steel thread```
+
+
+# Setup CHE to enable editing of the booster
+
+The Che Dashboard can be found at ```https://che-che.<ROUTER_SUFFIX>``` 
+
+## Self Signed Certs setup
+
+If using a self signed (non CA) certificate on the Che route, there are a few things that need to be done before a Workspace can be created and used.
+
+* Import the cert into Che Server (this is done autmatically by the ansible installer)
+* Create a custom Node.js stack with the self signed certificate included. This is required so that stack instances (workspaces) can make requests back to the Che server.
+* Import the self signed certificate into your browser. This is required so that a websocket connection back to the Che Server can be established, which is required by many parts of the Che Browser IDE.
+
+Docs for how to do items 2 & 3 are available at https://www.eclipse.org/che/docs/openshift-config.html.
+
+## Authorize Che against your Github Account
+
+Go to Profile > Preferences > SSH > VCS, and click the Github icon.
+This will trigger an OAuth flow in your browser and prompt you to login to Github and authorize Che.
+
+# Customise the messaging booster (optional)
+
+We can add some additional information to the message. To do this we will need to change some of the code in the messaging booster. 
+
+- Login to the Che Dashboard. The url is in the format ```https://che-che.<ROUTER_SUFFIX>``` 
+- Create a new workspace from the custom Node.js Stack (the stack that has the custom image with self-signed cert included)
+- Once the workspace is launched, click the `Workspace` menu item and select `Import Project`
```

Comment:
Think we need more detail here.

- Create a new workspace and select the custom Node.js Stack from the list of stacks
- Click Add or Import Project button
- Click the `Github` tab
- Select the messaging-frontend booster repo that was created by Launcher
- Click Create button at the bottom of the screen
-When prompted, click the 'Open in IDE' button